src/test/java/org/commcare/fixtures/test/IndexedFixtureTests.java

```diff
@@ -68,4 +75,48 @@ public void errorOnMaliciousSchemaTest() throws XPathSyntaxException, Unfullfill
             XmlPullParserException, IOException, InvalidStructureException {
         ParseUtils.parseIntoSandbox(getClass().getResourceAsStream("/malicious-indexed-fixture.xml"), sandbox, true);
     }
+
+    @Test
+    public void testPartialQueryLoads() throws XPathSyntaxException, UnfullfilledRequirementsException,
+            XmlPullParserException, IOException, InvalidStructureException {
+        ParseUtils.parseIntoSandbox(getClass().getResourceAsStream("/indexed_fixture/partial_lookup_load_body.xml"), sandbox);
+
+        EvaluationContext ec =
+                MockDataUtils.buildContextWithInstance(sandbox, "testfixture", "jr://fixture/testfixture");
+
+        EvaluationContext ecForTest = ec.spawnWithCleanLifecycle();
+
+        QueryContext context = ecForTest.getCurrentQueryContext();
+        ScopeLimitedReferenceRequestCache cache = context.getQueryCache(ScopeLimitedReferenceRequestCache.class);
+
+        String exprString = "instance('testfixture')/test/entry[@filter_attribute = 'pass'][true() and filter_one = 'pass']/name";
+        XPathExpression expr = XPathParseTool.parseXPath(exprString);
+
+        cache.addTreeReferencesToLimitedScope(new TreeReferenceAccumulatingAnalyzer(ecForTest).accumulate(expr));
+
+        CaseTestUtils.xpathEvalAndAssert(ecForTest, exprString, "succeed");
+    }
+
+    @Test
+    public void testPartialQueryLoadsInBulkMode() throws XPathSyntaxException, UnfullfilledRequirementsException,
```

Comment:
I'm gathering that this test is supposed to test a different code path from the first test. But I'm not exactly clear on what that disparate path is, and I also don't see anything that checks whether that code path was actually executed.

Reply:
There is no effective mechanism to accomplish that (ensuring that within the query framework that specific code paths were/were not triggered) currently, and it's a huge lift to build one, unfortunately. I think it's a pretty big chunk of tech debt to use the trace system to be able to detect what occurred during a query.